system/Publisher/ContentReplacer.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Publisher;
+
+class ContentReplacer
+{
+    /**
+     * Replace content
+     *
+     * @param array $replaces [search => replace]
+     */
+    public function replace(string $content, array $replaces): string
+    {
+        return strtr($content, $replaces);
+    }
+
+    /**
+     * Add text
+     *
+     * @param string $text    Text to add.
+     * @param string $pattern Regexp search pattern.
+     * @param string $replace Regexp replacement including text to add.
+     *
+     * @return bool|string true: already updated, false: regexp error.
```

Comment:
I find these return types pretty confusing. What about returning `?string` (`null` being "already added") and then throwing an exception for regex failure?

Reply:
I've changed return type completely.
Because `preg_replace()` returns `null` when error occurs.

>If matches are found, the new subject will be returned, otherwise subject will be returned unchanged or null if an error occurred. 
https://www.php.net/manual/en/function.preg-replace.php#refsect1-function.preg-replace-returnvalues